src/core/client/framework/lib/auth/auth.ts

```diff
@@ -0,0 +1,93 @@
+import { Claims, computeExpiresIn, parseAccessTokenClaims } from "./helpers";
+
+/**
+ * ACCESS_TOKEN_KEY is the key in storage where the accessToken is stored.
+ */
+const ACCESS_TOKEN_KEY = "coral:v1:accessToken";
+
+/**
+ * storage is the Storage used to retrieve/update/delete access tokens on.
+ */
+const storage = localStorage;
+
+export interface AuthState {
+  /**
+   * accessToken is the access token issued by the server.
+   */
+  accessToken: string;
+
+  /**
+   * claims are the parsed claims from the access token.
+   */
+  claims: Claims;
+}
+
+export type AccessTokenProvider = () => string | undefined;
+
+function parseAccessToken(accessToken: string) {
+  // Try to parse the access token claims.
+  const claims = parseAccessTokenClaims(accessToken);
+  if (!claims) {
+    // Claims couldn't be parsed.
+    return;
+  }
+
+  if (claims.exp) {
+    const expiresIn = computeExpiresIn(claims.exp);
+    if (!expiresIn) {
+      // Looks like the access token has expired.
+      return;
+    }
+  }
+
+  return { accessToken, claims };
+}
+
+export function retrieveAccessToken() {
+  try {
+    // Get the access token from storage.
+    const accessToken = storage.getItem(ACCESS_TOKEN_KEY);
+    if (!accessToken) {
+      // Looks like the access token wasn't in storage.
+      return;
+    }
+
+    // Return the parsed access token.
+    return parseAccessToken(accessToken);
+  } catch (err) {
+    if (process.env.NODE_ENV !== "production") {
```

Comment:
Just a question: What harm is there to output these errors in production?

Reply:
From a users perspective, it's not likely that this would be used to debug issues. If we're concerned about tracking these errors, we should consider these types of guards as indicators to include error reporters.